csharp/src/Google.Protobuf/MessageExtensions.cs

```diff
@@ -148,20 +148,33 @@ public static ByteString ToByteString(this IMessage message)
         /// </summary>
         public static bool IsInitialized(this IMessage message)
         {
-            if (message.Descriptor.File.Proto.Syntax != "proto2")
+            if (message.Descriptor.File.Proto.Syntax == "proto3")
```

Comment:
Looks like this could be invoked often. Should we introduce an enum property FileDescriptor.Syntax  that only gets initialized once and make an enum comparison here?